src/core/src/op/util/fft_base.cpp

```diff
@@ -38,164 +39,22 @@ void ov::op::util::FFTBase::validate() {
-        const auto input_rank = input_shape.rank().get_length();
-        const auto& const_axes = get_constant_from_source(input_value(1));
-        auto axes = const_axes->cast_vector<int64_t>();
-
-        // FFT operation supports for negative axes to transform. More precisely, according to
-        // the FFT operation specification, axes should be integers from -(r - 1) to (r - 2)
-        // inclusively, where r = rank(data). A negative axis 'a' is interpreted as an axis
-        // 'r - 1 + a'. The reason is the following: real input tensor of the shape
-        // [n_0, ..., n_{r - 1}, 2] is interpreted as a complex tensor with the shape
-        // [n_0, ..., n_{r - 1}].
-        for (int64_t& axis : axes) {
-            if (axis < 0) {
-                axis += input_rank - 1;
-            }
-        }
-
-        AxisVector axes_vector;
-        AxisSet axes_set;
-        for (const int64_t axis : axes) {
-            axes_vector.push_back(static_cast<size_t>(axis));
-            axes_set.insert(static_cast<size_t>(axis));
-        }
-
-        NODE_VALIDATION_CHECK(this, axes.size() == axes_set.size(), "FFT op axes must be unique. Got: ", axes_vector);
-
-        NODE_VALIDATION_CHECK(this,
-                              std::find(axes.begin(), axes.end(), input_rank - 1) == axes.end(),
-                              "FFT op axes cannot contain the last axis. Got axes: ",
-                              axes_vector);
-    }
-
     if (num_of_inputs == 3) {
         element::Type signal_size_et = get_input_element_type(2);
         NODE_VALIDATION_CHECK(this,
                               signal_size_et == element::i64 || signal_size_et == element::i32,
                               "FFT op signal_size element type must be i32 or i64");
-
-        const auto& signal_size_shape = PartialShape(get_input_partial_shape(2));
-        if (signal_size_shape.rank().is_static()) {
-            NODE_VALIDATION_CHECK(this,
-                                  signal_size_shape.rank().get_length() == 1,
-                                  "FFT op signal size input must be 1D tensor. Got signal size "
-                                  "input rank: ",
-                                  signal_size_shape.rank().get_length());
-        }
-
-        if (axes_shape.is_static() && signal_size_shape.is_static()) {
-            NODE_VALIDATION_CHECK(this,
-                                  axes_shape.to_shape()[0] == signal_size_shape.to_shape()[0],
-                                  "Sizes of inputs 'axes' and 'signal_size' must be equal. Got "
-                                  "size of 'axes': ",
-                                  axes_shape.to_shape()[0],
-                                  "size of 'signal_size': ",
-                                  signal_size_shape.to_shape()[0]);
-        }
     }
 }
 
 void ov::op::util::FFTBase::validate_and_infer_types() {
     NGRAPH_OP_SCOPE(util_FFTBase_validate_and_infer_types);
     validate();
```

Comment:
Is the method `validate()` used anywhere else besides this place? If not, can we remove it and move the element types validation to this method?

Reply:
validate() is only used in this function. Removed it.